src/views/MediaChannels/components/MoreInfoDrawer.tsx

```diff
@@ -19,37 +91,212 @@ interface MoreInfoDrawerProps {
-        <Typography
-          gutterBottom
-          variant="h5"
-          component="h2"
-          className={classes.title}
+      <div className={classes.close}>
+        <IconButton
+          aria-label={intl.formatMessage(messages.closeButton)}
+          component="span"
+          onClick={onClose}
         >
-          {product?.title}
-        </Typography>
-        <img
-          alt={product?.title}
-          src={product?.logo_rectangle_url?.[0]?.url}
-          className={classes.moreInfologo}
+          <CloseIcon />
+        </IconButton>
+      </div>
+      <div className={classes.content}>
+        <Grid container>
+          {!!duration?.period && (
+            <Grid item xs={6} className={classes.duration}>
+              <DateRangeIcon />
+              <Box>{` ${duration?.period} days`}</Box>
+            </Grid>
+          )}
+          <Grid item xs className={classes.price}>
+            <CreditCardIcon />
+            <Box>{` ${vonq_price?.[0].amount} ${vonq_price?.[0].currency}`}</Box>
+          </Grid>
+        </Grid>
+
+        <hr />
+
+        <div
+          className={classes.media}
+          style={{ backgroundImage: `url("${logo_rectangle_url?.[0]?.url}")` }}
+          title={title}
         />
-        <Typography gutterBottom variant="body1">
-          {product?.description}
+        <Typography gutterBottom component="div">
+          <Chip size="small" label={audience_group} />
+        </Typography>
+        <Typography variant="h5" component="h2">
+          {title}
+        </Typography>
+        {homepage && (
+          <Typography gutterBottom>
+            <Link
+              href={homepage}
+              target="_blank"
+              onClick={e => e.stopPropagation()}
+            >
+              {homepage}
+            </Link>
+          </Typography>
+        )}
+        <Typography className={classes.paragraph}>{description}</Typography>
+
+        <Typography variant="h6" component="h2">
+          <FormattedMessage {...messages.deliveryTime} />
+        </Typography>
+        <Typography className={classes.deliveryTime}>
+          <AccessTimeIcon fontSize="small" />
+          <span>
+            <FormattedMessage
+              {...messages.workingDays}
+              values={{ days: timeToProcess }}
+            />
+          </span>
         </Typography>
+        {maxTimeToProcess > timeToProcess && (
+          <Typography className={classes.deliveryTime}>
+            <span>
+              <FormattedMessage
+                {...messages.maxDeliveryTime}
+                values={{ maxTimeToProcess }}
+              />{" "}
+              <Link
+                href={intl.formatMessage(messages.learnWhyLink)}
+                target="_blank"
+              >
+                <FormattedMessage {...messages.learnWhy} />
+              </Link>
+            </span>
+          </Typography>
+        )}
+
+        {Array.isArray(cross_postings) && cross_postings.length > 0 && (
```

Comment:
Isn't this covered by `product` type?

Reply:
Not really, `product` is data fetch from external source at runtime. TS does not do runtime type validation unfortunately.